src/SingleArticle.jsx

```diff
@@ -0,0 +1,37 @@
+import { useParams } from 'react-router-dom';
+import { fetchArticleById } from './utils/api';
+import { useEffect, useState } from 'react';
+
+const SingleArticle = () => {
+      const [singleArticle, setSingleArticle] = useState([]);
+
+      const {article_id} = useParams();
+       useEffect(() => {
+              const getArticle = async () => {
+                try {
+                  const article = await fetchArticleById(article_id);
+                  setSingleArticle(article);
+                } catch (error) {
+                  console.error('Failed to load articles:', error);
+                }
+              };
+          
+              getArticle();
+            }, []);
```

Comment:
There is no indication of loading or error states which can lead to poor user experience.